main/src/main/MillServerMain.scala

```diff
@@ -183,33 +183,26 @@ class Server[T](lockBase: String,
     // flush before closing the socket
     System.out.flush()
     System.err.flush()
-
-    if (Util.isWindows) {
-      // Closing Win32NamedPipeSocket can often take ~5s
-      // It seems OK to exit the client early and subsequently
-      // start up mill client again (perhaps closing the server
-      // socket helps speed up the process).
-      val t = new Thread(() => clientSocket.close())
-      t.setDaemon(true)
-      t.start()
-    } else clientSocket.close()
```

Comment:
Where de we close the client socket now?

Reply:
Good catch, thx!  
I reverted it now. Although it doesn't do anything on windows: https://github.com/sbt/ipcsocket/blob/v1.0.1/src/main/java/org/scalasbt/ipcsocket/Win32NamedPipeSocket.java#L97

An aside: the ipcsocket library we use doesn't 100% follow the tutorial on MSDN docs.  
I tried to make an alternative implementation here https://github.com/sake92/Win32NamedPipe  
But then I realized it works even with ipcsocket as long as we ignore the exception when closing the socket in client...